test/src/test/java/hudson/model/labels/LabelExpressionTest.java

```diff
@@ -207,8 +210,10 @@ public void laxParsing() {
     @Test
     public void dataCompatibilityWithHostNameWithWhitespace() throws Exception {
         assumeFalse("Windows can't have paths with colons, skipping", Functions.isWindows());
-        DumbSlave slave = new DumbSlave("abc def (xyz) test", "dummy",
-                j.createTmpDir().getPath(), "1", Mode.NORMAL, "", j.createComputerLauncher(null), RetentionStrategy.NOOP, Collections.EMPTY_LIST);
+        DumbSlave slave = new DumbSlave("abc def (xyz) test", tempFolder.newFolder().getPath(), j.createComputerLauncher(null));
+        slave.setRetentionStrategy(RetentionStrategy.NOOP);
+        slave.setNodeDescription("dummy");
+        slave.setNodeProperties(Collections.emptyList());
```

Comment:
FTR one of the reasons for introducing the simplified constructor to begin with was to allow sensible defaults for most properties. At least `nodeDescription` and `nodeProperties` need not be overridden here, and the same is probably true of `retentionStrategy`.